### PR TITLE
Fix up experimental, add a makefile and add it to circle to stop experimental getting broken again.

### PR DESCRIPTION
**Pull request by tomwilkie**
Could also remove all the lead makefiles in experimental?

### CODE REVIEW

File: experimental/Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+all: $(targets)
+
+define T
+$1/$1: $1/*.go
+	go get -tags netgo ./$$(@D)
+	go build -ldflags "-extldflags \"-static\"" -tags netgo -o $$@ ./$$(@D)
+endef
+
+$(foreach d,$(dirs),$(eval $(call T,$d)))
```
Comment (peterbourgon) [inline experimental/Makefile]: Wow. Could we rather just have some some copy/pasted (dumb) targets?

File: experimental/Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+test:
+	go test ./...
+
+clean:
+	go clean ./...
```
Comment (peterbourgon) [inline experimental/Makefile]: Actually, can we not have a Makefile, please? Experimental stuff shouldn't be part of the regular build process.

File: circle.yml [hunk 3]
```diff
@@ -38,6 +38,7 @@ test:
     - cd $SRCDIR; make
     - cd $SRCDIR; ./bin/test
     - cd $SRCDIR; make client-test
+    - cd $SRCDIR/experimental; make
```
Comment (peterbourgon) [inline circle.yml]: Similarly I'd vote to remove that

**Comment by tomwilkie** [comment]
I added this as recent refactoring broken experimental, and a broken experimental makes automated refactoring tools (gorename) break.

File: experimental/graphviz/handle.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ import (
 func handleTXT(r Reporter) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
 		w.Header().Set("Content-Type", "text/plain")
-		dot(w, r.Report().Process.RenderBy(mapFunc(req), classView(req)))
+		//dot(w, r.Report().Process.RenderBy(mapFunc(req), nil, classView(req)))
```
Comment (peterbourgon) [inline experimental/graphviz/handle.go]: Can you fix this properly? If you don't want to bother, that's fine — in which case I think it's better to just temporarily remove graphviz from the Makefile or whatever.

File: experimental/Makefile (hunk 1 above)
Comment (peterbourgon) [inline experimental/Makefile]: I think this would be nicer if we just had individual, dumb, copy-pasted targets, removing dirs/targets/T/foreach

**Comment by peterbourgon** [comment]
:+1:

File: experimental/Makefile (hunk 1 above)
Comment (tomwilkie) [inline experimental/Makefile]: I can make it a little clearer without loosing the concision:

```
.PHONY: all test clean

DIRS=$(shell find . -maxdepth 2 -name *.go -printf "%h\n" | uniq)
TARGETS=$(join $(patsubst %,%/,$(DIRS)),$(DIRS))

all: $(TARGETS)

$(TARGETS):
	go get -tags netgo ./$(@D)
	go build -ldflags "-extldflags \"-static\"" -tags netgo -o $@ ./$(@D)
```

But it does loose the dependancies.

**Pull request merged**